commands/config.go

```diff
@@ -525,3 +535,34 @@ func configSubpathCmdF(cmd *cobra.Command, _ []string) error {
 
 	return nil
 }
+
+// cloudRestricted checks if the config path is restricted to the cloud
+func cloudRestricted(cfg any, path []string) bool {
+	t := reflect.TypeOf(cfg)
+	v := reflect.ValueOf(cfg)
+
+	if t.Kind() == reflect.Ptr {
+		t = reflect.Indirect(v).Type()
+		v = reflect.Indirect(v)
+	}
+
+	if t.Kind() == reflect.Struct {
+		for i := 0; i < t.NumField(); i++ {
+			field := t.Field(i)
+			if len(path) == 0 || field.Name != path[0] {
+				continue
+			}
+
+			if field.Type.Kind() == reflect.Struct {
+				return cloudRestricted(v.Field(i).Interface(), path[1:])
```

Comment:
It seems strange to pack this back into an interface just to reflect it in the next recursion. A separate function for the recursion would solve that.

Reply:
Makes sense, updated to use `reflect.Type` all the time. It's much more convenient now. Thanks.